src/Core/Resolvers/MsSqlQueryExecutor.cs

```diff
@@ -305,12 +305,51 @@ public override string GetSessionParamsQuery(HttpContext? httpContext, IDictiona
             return dbResultSet;
         }
 
+        /// <inheritdoc />
+        public override SqlCommand PrepareCommand(
+            SqlConnection conn,
+            string sqltext,
+            IDictionary<string, DbConnectionParam> parameters,
+            HttpContext? httpContext,
+            string dataSourceName)
+        {
+            SqlCommand cmd = conn.CreateCommand();
+            cmd.CommandType = CommandType.Text;
+
+            // Add query to send user data from DAB to the underlying database to enable additional security the user might have configured
+            // at the database level.
+            string sessionParamsQuery = GetSessionParamsQuery(httpContext, parameters, dataSourceName);
+
+            cmd.CommandText = sessionParamsQuery + sqltext;
+            if (parameters is not null)
+            {
+                foreach (KeyValuePair<string, DbConnectionParam> parameterEntry in parameters)
```

Comment:
Odd behavior seen in debugger: when you submit a GraphQL request like the following:
```graphql
query GetBooksPaged {
  books(first: 5) {
    items {
      id
      title
      publisher_id
    }
    endCursor
    hasNextPage
  }
}
```

the generated sql query looks like:
```tsql
SELECT TOP 6 [table0].[id] AS [id], [table0].[title] AS [title], [table0].[publisher_id] AS [publisher_id] FROM [dbo].[books] AS [table0] WHERE 1 = 1 ORDER BY [table0].[id] ASC FOR JSON PATH, INCLUDE_NULL_VALUES
```

yet the variable `parameters` has three entries. So there is work being done here for parameters that aren't even in the generated sql query. 

Only reason i saw this is because when creating `SqlParameter parameter = cmd.CreateParameter();` you can use the `SqlParameter` constructor which takes in a `SqlDbType`. If you provide that , then `SqlParameter` automatically resolves the `DbType` rendering the helper function `PopulateDbTypeForParameter(..)` as no longer needed.

Reply:
While the issue i mentioned may not necessarily need to be addressed in this pr, please file an issue for it so we can track. 

Here are the three params that are present:
![image](https://github.com/Azure/data-api-builder/assets/6414189/0117cfd9-d783-49c6-8e62-d7c994acb566)

Reply:
created an issue #2125 to track the same. Thanks for pointing out.